ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/embeddedActivityBlock/embeddedActivityBlock.component.ts

```diff
@@ -1,62 +1,108 @@
-  @Input() instance: ActivityInstance;
-  @Input() readonly: boolean;
-  @Input() validationRequested: boolean;
-  @Input() studyGuid: string;
-  @Output() componentBusy = new EventEmitter<boolean>(true);
-
-  public activity: ActivityForm;
-  private readonly LOG_SOURCE = 'EmbeddedActivityBlockComponent';
-
-  constructor(private activityServiceAgent: ActivityServiceAgent,
-              private changeDetector: ChangeDetectorRef,
-              private submitAnnouncementService: SubmitAnnouncementService,
-              private logger: LoggingService) { }
-
-  ngOnInit(): void {
-    this.getActivity();
-    this.submitAnnouncementService.submitAnnounced$.pipe(
-        tap(() => this.componentBusy.emit(true)),
-        concatMap(() => this.activityServiceAgent.flushForm(this.studyGuid, this.instance.instanceGuid)),
-        catchError(err => {
-            this.logger.logError(this.LOG_SOURCE, 'An error during completing an activity', err);
-            return EMPTY;
-        }),
-        take(1)
-    ).subscribe(() => this.componentBusy.emit(false));
-  }
-
-  public getSectionId(index: number, {name}: ActivitySection): string {
-    return `${index}_${name}`;
-  }
-
-  private getActivity(): void {
-    this.activityServiceAgent.getActivity(of(this.studyGuid), of(this.instance.instanceGuid))
-      .pipe(
-        catchError(err => {
-          this.logger.logError(this.LOG_SOURCE, 'An error during getting a full activity', err);
-          return EMPTY;
-        }),
-        take(1))
-      .subscribe((activity: ActivityForm) => {
-        this.activity = activity;
-        this.changeDetector.detectChanges();
-      });
-  }
+    @Input() instance: ActivityInstance;
+    @Input() readonly: boolean;
+    @Input() validationRequested: boolean;
+    @Input() studyGuid: string;
+    @Output() componentBusy = new EventEmitter<boolean>(true);
+    @Output() deletedActivity = new EventEmitter<string>();
+    @ViewChild('delete_button', {read: ElementRef}) private deleteButtonRef: ElementRef;
+
+    public activity: ActivityForm;
+    private readonly LOG_SOURCE = 'EmbeddedActivityBlockComponent';
+
+    constructor(private activityServiceAgent: ActivityServiceAgent,
+                private changeDetector: ChangeDetectorRef,
+                private submitAnnouncementService: SubmitAnnouncementService,
+                private logger: LoggingService,
+                private dialog: MatDialog,
+                private modalService: ActivityBlockModalService) {
+    }
+
+    ngOnInit(): void {
+        this.getActivity();
+        this.flushFormOnSubmit();
+    }
+
+    public getSectionId(index: number, {name}: ActivitySection): string {
+        return `${index}_${name}`;
+    }
+
+    public openDeleteDialog(): void {
+        const config = this.modalService.getDeleteDialogConfig(this.deleteButtonRef);
+        config.data = {
+            actionCallback: this.deleteInstance.bind(this)
```

Comment:
optional: instead of passing callback into the dialog consider listening to afterClosed observable and make an action based on the returned by the dialog result.
you can find examples here https://material.angular.io/components/dialog/examples
advantages: all reactive approach/observables advantages vs callbacks.